pmdarima/utils/tests/test_array.py

```diff
@@ -135,7 +135,7 @@ def test_diff():
 )
 def test_diff_inv(arr, lag, differences, xi, expected):
     res = diff_inv(arr, lag=lag, differences=differences, xi=xi)
-    expected = np.array(expected, dtype=np.float)
+    expected = np.array(expected, dtype=float)
```

Comment:
OOC what is the impetus for these dtype changes?

Reply:
The impetus on our side were the build warnings like this:

```
DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    frec = np.ones(int((s + 1) / 2), dtype=np.int)
```

On the [numpy side](https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations) "naked" dtypes seem to have been aliases for a while and just because deprecated. So to get rid of the warning the options were to just use the Python built-in or to add a precision (with `np.bool` only having one option, obviously).

Basically I am just trying to stay on top of warnings in case things ever get removed. That has bitten us in the past 🤷🏻